### PR TITLE
Added trace log level

### LINKED ISSUE #1547: Add a Trace log level
## Feature Request

Trace log level.

#### Overview:

We should add a new "Trace" log level that would be used for things that needs a high amount of logs, and would be normally undesirable to enable unless the person is trying to diagnose a specific issue.

Example of things that should use such "Trace" level are the kernel syscall logs and IPC call logs.

### Why would this feature be useful?

This will make the debug log level more useful, as we can print some debug information with it without slowing down the guest applications a lot and without spamming the console and generating a huge log file.

### PR DESCRIPTION
**Pull request by mlgatto**
Added support for trace logs - fixes #1547

### CODE REVIEW

**Comment by gdkchan** [comment]
Can you also change:
https://github.com/Ryujinx/Ryujinx/blob/master/Ryujinx.HLE/HOS/Kernel/SupervisorCall/SyscallTable.cs#L482
and:
https://github.com/Ryujinx/Ryujinx/blob/master/Ryujinx.HLE/HOS/Services/IpcService.cs#L120
To use the new trace log level instead of debug?

File: Ryujinx.Common/Logging/Logger.cs [hunk 1]
```diff
@@ -196,6 +198,7 @@ public static void SetEnable(LogLevel logLevel, bool enabled)
                 case LogLevel.Guest     : Guest     = enabled ? new Log(LogLevel.Guest)    : new Log?(); break;
                 case LogLevel.AccessLog : AccessLog = enabled ? new Log(LogLevel.AccessLog): new Log?(); break;
                 case LogLevel.Stub      : Stub      = enabled ? new Log(LogLevel.Stub)     : new Log?(); break;
+                case LogLevel.Trace     : Trace     = enabled ? new Log(LogLevel.Trace) :    new Log?(); break;
```
Comment (AcK77) [inline Ryujinx.Common/Logging/Logger.cs:201]: Alignment is off here

**Review by AcK77 — CHANGES_REQUESTED** [review]
You have to update the config file to add the new field, even if it’s disabled as default!

**Comment by Pixxel123** [comment]
If it's not going to be enabled by default, I'd probably move the UI toggle to be at the bottom of the list, after Guest Logs and before Fs Access Logs to make it easier for users to have just the defaults on.

**Comment by mlgatto** [comment]
> You have to update the config file to add the new field, even if it’s disabled as default!

@AcK77 Where can I find the config file?

**Comment by gdkchan** [comment]
> > You have to update the config file to add the new field, even if it’s disabled as default!
> 
> @AcK77 Where can I find the config file?

This shouldn't be needed anymore, the default config file was removed some months ago.

**Comment by mlgatto** [comment]
> I mean you still have to update the code which update the config file is needed, see https://github.com/Ryujinx/Ryujinx/blob/master/Ryujinx/Configuration/ConfigurationState.cs#L978 as example.

@AcK77 sorry I'm a bit lost, I think I already have updated that section of the code. Can you elaborate a bit more what is still needed? Thanks :)

**Comment by jduncanator** [comment]
> > I mean you still have to update the code which update the config file is needed, see https://github.com/Ryujinx/Ryujinx/blob/master/Ryujinx/Configuration/ConfigurationState.cs#L978 as example.
> 
> @AcK77 sorry I'm a bit lost, I think I already have updated that section of the code. Can you elaborate a bit more what is still needed? Thanks :)

You need to add a migration from version 35 to your new version. Even though the migration doesn't really need to do anything explicitly (the default value of a bool is false after all), you still need to do a migration to trigger Ryujinx to reserialize the configuration file on first load, bump the version number, and add the new trace log level option to the configuration file.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by Thog — APPROVED** [review]

**Review by riperiperi — COMMENTED** [review]
Nice, will be good having a trace log level. I don't see anything in here to update the dialog that appears when trying to start a game with Debug logs, though. 

The UI has a dialog that pops up and essentially asks the user to disable them due to potential performance impact - but the majority of these logs have been moved to the trace level. I think the warning should be instead for that level, and also the placement of the checkbox in settings might be worth moving next to debug too.

**Comment by AcK77** [comment]
Could you please address the comments of Riperiperi then we can merge it ;)!

**Review by riperiperi — APPROVED** [review]

**Pull request merged**